app/javascript/packages/modal/modal-element.ts

```diff
@@ -0,0 +1,43 @@
+import { createFocusTrap } from 'focus-trap';
+import type { FocusTrap } from 'focus-trap';
+
+class ModalElement extends HTMLElement {
+  trap: FocusTrap;
+
+  connectedCallback() {
+    this.trap = createFocusTrap(this, { escapeDeactivates: false });
+    this.addEventListener('click', this.#handleDismiss);
```

Comment:
QQ: is this because the event listeners get called with the wrong `this`? what if we made it an arrow function since those don't get their own `this`?

```suggestion
    this.addEventListener('click', (event) => this.handleDismiss(event));
```

Reply:
Yeah, I flip-flop between these two different approaches, since I haven't come to a strong preference one way or the other. They're largely the same.

And yes, the reason we'd need it is since `this` would be incorrectly bound without defining as instance initializer.

The `#` is a separate thing, and also not something I feel too strongly about, but thinking about what we'd want to be the public interface of the element, this felt more of a private member.